File: frontend/src/lib/currency.ts
const phpCurrency = new Intl.NumberFormat('en-PH', {
  style: 'currency',
  currency: 'PHP',
  maximumFractionDigits: 0,
  minimumFractionDigits: 0,
});

export function formatPHP(value: number): string {
  if (!Number.isFinite(value)) return phpCurrency.format(0);
  return phpCurrency.format(value);
}

File: mobile/lib/core/utils/currency.dart
import 'package:intl/intl.dart';

/// PHP currency formatter, mirroring `frontend/src/lib/currency.ts`.
final NumberFormat _phpFormat = NumberFormat.currency(
  locale: 'en_PH',
  symbol: '₱',
  decimalDigits: 0,
);

String formatPHP(num? value) {
  final v = value ?? 0;
  if (v.isNaN || v.isInfinite) return _phpFormat.format(0);
  return _phpFormat.format(v);
}
